scripts/system_performance/cycle_webscraper.py

```diff
@@ -0,0 +1,198 @@
+# ############################################################################### #
+# Autoreduction Repository : https://github.com/ISISScientificComputing/autoreduce
+#
+# Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI
+# SPDX - License - Identifier: GPL-3.0-or-later
+# ############################################################################### #
+
+"""
+The purpose of this package is for performing MySQL queries to monitor system state
+performance and health. This file retrieves and cleans cycle data for
+use in MySQL queries.
+"""
+
+# Dependencies
+# Web-scraping dependencies
+import os
+import logging
+import re
+import datetime
+import requests
+import lxml.html as lh
+import pandas as pd
+
+
+class TableWebScraper(object):
+    """
+    Web-scrapes beam cycle data from https://www.isis.stfc.ac.uk/Pages/Beam-Status.aspx
+    for MySQL system checks. This URl can be changed at the bottom of the script
+    """
+
+    def __init__(self, url, local_data=None):
+        self.url = url
+        if not local_data:
+            # By default, a local copy of cycle data is stored in the current working directory
+            self.local_data = "cycle_dates.csv"
+        else:
+            # Allows for a custom location to be set for a local CSV copy of cycle data
+            self.local_data = local_data
+
+    def save_to_csv(self, data):
+        """ Create local copy of dataframe as .csv to read when web page is not accessible"""
+        data.to_csv(r'{}'.format(self.local_data), encoding='utf-8')
+        return data
+
+    @staticmethod
+    def read_csv(local_data):
+        """Reads csv into a pandas data frame """
+        if os.path.isfile("{}".format(local_data)):
```

Comment:
What type is local_data here? I would suggest that perhaps `os.path.isfile(str(local_data)):` would be cleaner